github/repos.go

```diff
@@ -509,37 +509,49 @@ type Branch struct {
+	Restrictions               *BranchRestrictions            `json:"restrictions"`
 }
 
 // ProtectionRequest represents a request to create/edit a branch's protection.
 type ProtectionRequest struct {
-	RequiredStatusChecks       *RequiredStatusChecks       `json:"required_status_checks"`
-	RequiredPullRequestReviews *RequiredPullRequestReviews `json:"required_pull_request_reviews"`
-	EnforceAdmins              bool                        `json:"enforce_admins"`
-	Restrictions               *BranchRestrictionsRequest  `json:"restrictions"`
+	RequiredStatusChecks       *RequiredStatusChecks                 `json:"required_status_checks"`
+	RequiredPullRequestReviews *PullRequestReviewsEnforcementRequest `json:"required_pull_request_reviews"`
+	EnforceAdmins              bool                                  `json:"enforce_admins"`
+	Restrictions               *BranchRestrictionsRequest            `json:"restrictions"`
 }
 
 // RequiredStatusChecks represents the protection status of a individual branch.
 type RequiredStatusChecks struct {
-	// Enforce required status checks for repository administrators. (Required.)
-	// Deprecated: Use EnforceAdmins instead.
-	IncludeAdmins bool `json:"include_admins"`
 	// Require branches to be up to date before merging. (Required.)
 	Strict bool `json:"strict"`
 	// The list of status checks to require in order to merge into this
 	// branch. (Required; use []string{} instead of nil for empty list.)
 	Contexts []string `json:"contexts"`
 }
 
-// RequiredPullRequestReviews represents the protection configuration for pull requests.
-type RequiredPullRequestReviews struct {
-	// Enforce pull request reviews for repository administrators. (Required.)
-	// Deprecated: Use EnforceAdmins instead.
-	IncludeAdmins bool `json:"include_admins"`
+// PullRequestReviewsEnforcement represents the pull request reviews enforcement of a protected branch.
+type PullRequestReviewsEnforcement struct {
+	DismissalRestrictions *DismissalRestrictions `json:"dismissal_restrictions"`
+	DismissStaleReviews   bool                   `json:"dismiss_stale_reviews"`
+}
+
+// PullRequestReviewsEnforcementRequest represents request to set the pull request review
+// enforcement of a protected branch. It is separate from PullRequestReviewsEnforcement above
+// because the request structure is different from the response structure.
+type PullRequestReviewsEnforcementRequest struct {
+	DismissalRestrictionsRequest *DismissalRestrictionsRequest `json:"dismissal_restrictions"`
+	DismissStaleReviews          bool                          `json:"dismiss_stale_reviews"`
+}
+
+// PullRequestReviewsEnforcementPatchRequest represents request to patch the pull request review
+// enforcement of a protected branch. It is separate from PullRequestReviewsEnforcementPatchRequest above
+// because the patch request does not require all fields to be initialized.
+type PullRequestReviewsEnforcementPatchRequest struct {
```

Comment:
The HTTP method is indeed `PATCH`, but that's a low level implementation detail. The endpoint is called "Update pull request review enforcement of protected branch", which is where my suggestion of using "Update" comes from.

It's still not ideal, because "Request" is used for an endpoint called "Update branch protection"... But still, I think one word is better than "PatchRequest".